runtime/doc/fold.txt

```diff
@@ -148,6 +142,22 @@ end in that line.
 It may happen that folds are not updated properly.  You can use |zx| or |zX|
 to force updating folds.
 
+Minimizing Computational Cost			             *fold-expr-slow*
+
+Due to its computational cost, this fold method can make Vim unresponsive,
+especially when the fold level of all lines have to be initially computed.
+Afterwards, after each change, Vim restricts the computation of foldlevels
+to those lines whose fold level was affected by it (and reuses the known
+foldlevels of all the others).
+
+The foldexpr should therefore strive to minimize the number of dependent lines
+needed for the computation of a given line: For example, try to avoid the "=",
+"a" and "s" return values, since Vim often has to search backwards for
+a line for which the fold level is defined.
```

Comment:
Perhaps this is better?

```diff
needed for the computation of a given line: For example, try to avoid the "=",
-"a" and "s" return values, since Vim often has to search backwards for
-a line for which the fold level is defined.
+"a" and "s" return values, because these will require the evaluation of the
+fold levels on previous lines until an independent fold level is found.
```